### ISSUE TITLE
2.4.0 won't open to GUI with wifi enabled:  "A network error occurred"

### ISSUE THREAD

**Issue by mgoldeyy** [ultrasonic/ultrasonic#203]
## Problem description

Since the upgrade to 2.4.0 the app opens to an error message whenever wifi is enabled on my phone.  

The error is:  "Error.  A network error occurred.  Please check the server address or try again later."

It is not possible to access the GUI and use the application.  Turning off wifi solves this problem and the app starts normally.

This problem did not exist until my Ultrasonic icon changed about a week ago, so I believe it is related to the latest release.

### Steps to reproduce

1) Turn on wifi on a properly configured Ultrasonic app on an Android device
2) Restart the app.

## System information

 * **Ultrasonic version**: 2.4.0
 * **Android version**: 7.0 security patch through 3/2018
 * **Device info**:  S6 

## Additional notes

At first, I thought this might be an issue with having an Internet routable address for the Subsonic server when connected to a non-Internet routable LAN via wifi (which takes precedence over my phone carrier's network).  I say that because, historically, connecting to my home wifi and specifying my home server's public IPv4 address eventually resulted in the same(?) "network error" message as above.   This is likely due to the configuration of my routers.  

But now, I have the same issue when I change the server address to a LAN IP address, which I created long ago so that I could use Ultrasonic while at home on wifi.  

Also, the message comes up immediately instead of after the app starts trying to access the server for content.  This makes the app unusable -- there's no way to change settings or select a different server because the app is inaccessible.

**Comment by ogarcia** [comment]
@mgoldeyy if the app works well with phone network and not with WiFi then is a misconfiguration. Take note that the app is agnostic to connection, it only uses the available connection.
You can try to create a new connection on settings with your data. If not work make sure that you can access with same data via browser (Chrome or similar).

**Comment by Tapchicoma** [comment]
@mgoldeyy most likely app can't access server on WIFI network. Though not allowing using the app is definitely a bug.

**Comment by mgoldeyy** [comment]
Thanks for the feedback.  

I think that the real bug is not allowing the app to run and closing it before the user can change any settings, and I think that this may be a change in when Ultrasonic checks its network settings.

This morning, I wanted to see if the app would work using wifi and a LAN server address if I started it with wifi turned off.  It does start, which is different from yesterday.  But, then, if I switch servers to the public IP address, I get the "network error" message.  That's expected, because there is no route between the wifi network and the WAN IP address of my server.  

I cannot explain why Ultrasonic is working differently this morning.  But I do think that it has something to do with the upgrade to 2.4.0.  Here is a little more information:

1) I have been running my Subsonic / Ultrasonic setup for 2 years or longer, no problems and no changes to anything (except at the last upgrade which added an option to ignore self-signed SSL certificates or something like that).

2) I rebooted my phone in the middle of all of this to make sure it wasn't an Android system error.  Rebooting didn't change Ultrasonic's behavior.

3) If I start the app with wifi on and then change the settings to use the public IP, I get the "a network error" message ***but*** I can then click OK and continue to use the app, and change settings.  

4) Sometimes, when #3 happens, clicking "OK" seems to close the app.  When I tap its icon, it comes back up again with the error message displayed on the screen.  Yesterday, the app was definitely closing.  Today, sometimes, it seems like the app window is being destroyed by the app is still running.

Hope this helps.

The difference is that the error also appears at startup, which it never did before.  As noted, there's no way to use the app after clearing the error message.
